rust/gui-client/src-tauri/src/client/network_changes/windows.rs

```diff
@@ -112,54 +112,111 @@ pub fn check_internet() -> Result<bool> {
     Ok(have_internet)
 }
 
-/// Worker thread that can be joined explicitly, and joins on Drop
+/// Worker thread (on Windows) that can be joined explicitly, and joins on Drop
 pub(crate) struct Worker {
     inner: Option<WorkerInner>,
     rx: mpsc::Receiver<()>,
 }
 
 /// Needed so that `Drop` can consume the oneshot Sender and the thread's JoinHandle
 struct WorkerInner {
-    thread: std::thread::JoinHandle<Result<(), Error>>,
-    stopper: tokio::sync::oneshot::Sender<()>,
+    thread: thread::JoinHandle<Result<()>>,
+    shutdown_tx: oneshot::Sender<()>,
+}
+
+pub(crate) fn dns_notifier(tokio_handle: tokio::runtime::Handle) -> Result<Worker> {
+    Worker::new("DNS listener", move |tx, shutdown_rx| {
+        // Do `block_on` so we use our own worker thread
+        tokio_handle.block_on(async move {
```

Comment:
I think this can be simplified a bit if I understand `LocalSet` correctly:

- Construct the `LocalSet` directly on the new thread.
- Use `LocalSet::spawn_local` for the DNS task
- Call `handle.block_on` with the `LocalSet` as a value (it will only resolve until all its tasks have exited)

Reply:
Looks like if I await the whole `LocalSet` I can't get the result from the task. But I think in the upcoming PR where I redo this, it is a little simpler.